### PR TITLE
Refactoring ImageFeature for TF 2.0

### PR DESCRIPTION
**Pull request by ydudin3**
Tested on MNIST dataset with the following model 

input_features:
    -
        name: image_path
        type: image
        preprocessing:
            in_memory: true

output_features:
    -
        name: label
        type: numerical

training:
    dropout_rate: 0.4

Using numerical outputs for now since categories are not implemented yet.

Added ImageTestEncoder which uses FCStack only (Conv stacks are not implemented yet) just to test end-to-end training.

### CODE REVIEW

File: ludwig/models/modules/image_encoders.py [hunk 1]
```diff
@@ -14,10 +14,43 @@
...
+
+    def call(self, inputs, training=None, mask=None):
+        """
+            :param inputs: The inputs fed into the encoder.
+                   Shape: [batch x height x width x channels], type tf.uint8
+        """
+
+        # use FCStack only for now as it is already implemented and 
+        # ConvStack2D is not
+        fc_stack = FCStack(
```
Comment (w4nderlust) [inline ludwig/models/modules/image_encoders.py:41]: This goes into the init

Comment (ydudin3) [reply ludwig/models/modules/image_encoders.py]: Right.. updated.

**Comment by w4nderlust** [comment]
looks good so far, next step is to implement the encoders.
You can look at FCStack as an example of an encoder that was adapted to TF2.
Also I'm pretty sure that online you can find several examples of both VGG and ResNet for TF2/Keras. I'm also pretty sure they are also available as example models already in their codebase.

It could also be a good time to look into loading pretrained models if you have time as with TF2 it should be substantially easier.

**Comment by ydudin3** [comment]
Let's merge this to tf_porting and I'll follow up with new PRs? I generally try to discourage commits which modify more than a couple hundred lines of code.
